### ISSUE TITLE
Seek user review in a polite unobtrusive manner

### ISSUE THREAD

**Issue by GaurangTandon** [GaurangTandon/ProKeys#36]
![image](https://cloud.githubusercontent.com/assets/6308683/17025837/579a42a0-4f7b-11e6-8705-9140e4562af0.png)

Utilize either of the above two spaces to ask the user's a question (after ProKeys usage of about a week): **"Hey! How has your experience been so far?"** - with two options - **"Awesome!"/"Okayish"**.

1. Clicking awesome would convert the box to "Wonderful! Tell the world about it!" with two options - "Yes!" (opens the chrome store)/"Not now"
2. Clicking Okayish would convert the box to "We're sorry! Please tell us what went wrong" with two options - "Yes!" (opens a survey form)/"Not now"

**Points to decide:**

1. UI of the popup box (@iwrote your domain)
2. What to do in case of clicking "not now".

**Refer to**: Different tuts on "unobtrusive user feedback"

**Comment by iWrote** [comment]
## Okay. I think this (kinda) works.
I was thinking ProKeys is all about text shortcuts. I want to begin incorporating a tiny bit of theme into the UI. And I realised that we have polyglots and ILR1,2,3 (i.e. non-native anglophones) users. So I decided to follow the Webstore's guidelines about minimising text and some decent advice [here](https://uxmag.com/articles/the-psychologists-view-of-ux-design).

## Solution (Might wanna enlarge images.)
### After x days of using prokeys.
This happens.
![after-x-days](https://cloud.githubusercontent.com/assets/18333895/18996660/c9e8dbf4-874e-11e6-8cc4-ac318c3405cc.png)
The sudden appearance of an extra button. A lowercase f breaks the pattern with the other buttons. And its resemblance to a certain book is also very catchy. 


### Case 1 of 2: User Ignores Button
That's alright. They'll click it someday. If a user ignores a new button, then I doubt a message would do a lot of good. But, I guess we could, after x **+ y** days:
![thickheaduserfeedbackprompt](https://cloud.githubusercontent.com/assets/18333895/18998910/42332606-8758-11e6-8c7a-94ffdfbb8bf6.png)
Display a message and an asterisk (circular). This is for the thickheaded users **only**. Meant for use when the button goes unclicked for a very long time.

## Case 2 of 2: User Clicks Button
Yay! But is this doable?
![after-click](https://cloud.githubusercontent.com/assets/18333895/18998984/b6661178-8758-11e6-9196-6627af7cee29.png)
I used text emoticons as they're universal across languages. And I googled upright dongers so that even the less savvy users can understand them. I felt text icons might be easier to implement as well.
The buttons below them are supposed to work like the old [Change Hotkey] button in v2(.7? idk). I remember it was an outline which sharply filled itself with colour onMouseHover.

The sudden expansion of the button automatically asks the question.

### MEH = Okayish
Leads the user to the ProKeys Feedback form. I made some edits [there](https://docs.google.com/forms/d/e/1FAIpQLSehhUTJ198quynfGdzzWMgUpvO1dWtwvwyAfOIfKCxIOgBXXw/viewform) to incorporate the bug report into a feedback form. I've set up the conditions with one assumption: **Bugs are easier to handle in formatted reports; Questions and Suggestions should be redirected the 'Tell the Developer' section on the Chrome Webstore page."**

### NICE = Awesome
Opens the reviews page in a new tab.

### Red Cross = Not Now
Intuitive close button. Collapses the button.

## Guidelines
### First-time use.
- Button appears after x days.
- Clicking the feedback button for the first time stops the reminder-message-timer permanently.
- feedback button does not highlight green in expanded state.
- Clicking MEH or NICE collapses the button and opens relevant link in new tab.
- Red cross closes the window. It's equivalent to Not Now. So follow the First-time-use guidelines when feedback is clicked again.

### After first-time use (i.e. user has clicked NICE or MEH once in a lifetime)
- The button stays. User might wish to submit additional reports later.
- Clicking MEH acts as it did before.
- Clicking NICE just displays a Thanks! alert pop-up. 

## My Questions/Issues 

1. Is it doable?
2. Can you check whether a particular user has written a review or not? I have assumed it can't be done. Otherwise, the behaviour of the button can be improved.
3. Modern fonts are proportional, and kern. So displaying the text emojis properly might present difficulties. 
4. Anti-spam measures on the Google Form? The feedback button makes it very easy for the user to just hit MEH and submit a whining complaint before even checking the FAQ.

**Comment by GaurangTandon** [comment]
Excellent! I like the emoji idea and the new form. Here are my first concerns:

1. The feedback button in lowercase _is_ a good idea. **But**, why is it in a dull colour? I am not an avid reader of a book, and it doesn't give the appearance of a book in any manner :/
2. As a user, if clicking "Nice" suddenly exposed me to a public page, I'd be scared. I would have expected the dialog to just say "Thanks!" and go away. So, it should instead be something like "Wonderful! Tell the world about it" or "Awesome! Leave us a review." and then it should say "not now" or "yeah sure", and _then_ lead to the public page. Same goes with the "Meh" button. It should lead to the form after displaying a message stating it's going to do so.
3. Certain users with smaller screens have ProKeys in a different layout like this (resize your own window to see the effect):

![image](https://cloud.githubusercontent.com/assets/6308683/19011219/280e0098-87ad-11e6-9a07-bd201880467b.png)

Where to show the dialog box there?

-------------

Now, replying to **your questions:**

1. It certainly is :)
2. Nope. Though what change would you have done if the user had written a review (and we could have detected it)?
3. We can use images of emojis. Not a big problem.
4. We can add an additional section, that is shown before the bug report section, having a question with options like "I have read all the troubleshooting tips in the Help page and they didn't help." and "I didn't even know they existed. Going to check right now!"

---------

P.S. You were talking of this button?
Before hover:
![step1b](https://cloud.githubusercontent.com/assets/6308683/19011254/4379c24e-87ae-11e6-991c-6ef3c6ecb0ab.PNG)
After hover:
![step2b](https://cloud.githubusercontent.com/assets/6308683/19011255/46299514-87ae-11e6-9956-4579d1f7fc45.PNG)  

P.P.S. Can't add everything to v3.1.0 :P So, have pushed this to v3.2.0.

**Comment by iWrote** [comment]
##Reply to 1,2,3
 1.  facebook  

I thought it was close. I was so happy with it. Only one thing left to do. A pistol keeps faith . . .
😢 🔫  

P.S. I agree it looks dull in the image. I just meant a thinner font. Calibri Light (also very mainstream, shouldn't cause issues.) And I think it was accidentally faded due to wrong tolerance setting of the Fill Tool.

 2.
MEH opens popup - "Open feedback form?" - Yes/Later   OR
                                  "Help us improve! Open feedback form?" - Yes/Later

NICE opens popup -   "Write a review?" - Yes/Later       OR
                                    "Help us grow! Write a review?" - Yes/Later

The longer sentences might not look good in a Chrome popup, but they make more sense.

**Three**. (Git is messing with my numbers.)  Yeah, I know. I use it in high zoom myself. I thought it works, lots of whitespace around, I just didn't feel like drawing it because it required a lot of alignment, cut-crop-paste, proportions. 
I've done it now. It's slightly out of proportion, ignore that.
![zoomed in](https://cloud.githubusercontent.com/assets/18333895/19014106/93fe2e14-8801-11e6-8f18-f3b477272bd1.png)



##  Reply Recursion. In replying to your replying 😆 
 **Two.** What if the user does click Yes after clicking NICE. **But does not actually write a review.** 
After that the NICE button just says thanks. 

Ideally, the NICE button retains its linking function until the user actually writes a review, then it just thanks them.

### Possible Solutions:
a. You could always redirect to reviews page, but it's meaningless for someone who already has a review. 

b. You could just ask them if they have already written a review on the Chrome Webstore. 
**If a popup is limited to 2 buttons.**
Clicking "Later" in the Write a Review? popup leads to another popup "Have you already written a review on the Chrome Webstore?" - Yes/No
**If a popup can have 3 buttons and still look clean.**
NICE = "Help us grow! Write a Review?" - Yes / Later / Already Have
("Already Have" could be better worded WIP. Used longer sentence here because 3 buttons will naturally give us more space.)
MEH works as it did before.

c. We can scrap this whole idea. I really do not mind. All new ideas present new issues. But I really do NOT mind scrapping. 

 **Three.** Nice. Disproportionating fonts is very bothersome. Images are easy.

  **Four.** Revised the form. 
I also tried merging all Yes/No questions into checkboxes where they made more sense. But it didn't quite work.

**Comment by iWrote** [comment]
I'm sorry. For some reason Git made my numbering go haywire! Lists, sub-lists, sub-sub-lists, indents.

**Comment by GaurangTandon** [comment]
## Reply to 1,2,3

1. Ummm, it really wasn't anywhere close to facebook :/ Of that dullness, ok fine, will use a better font.
2. Couldn't clicking "Meh" simply remove the two emojis and replace them with "Help us improve! Open feedback form? - Yes/Later" Same thing for "Nice" I think with the emojis removed we can fit these long sentences without the need for a popup.
3. Good UI +1

# Reply Recursion. In replying to your replying to my replies 😆

**Two**
If the user doesn't write a review, we can't detect that and I think we don't need to care for it. The user himself said "yes" to write the review after hitting "Nice" emoji. 

> a. You could always redirect to reviews page, but it's meaningless for someone who already has a review.

So, he'll close the reviews tab if he's already written the review, simple. And since he did click Yes, he won't get a feedback notification again. What's the problem?

> b. You could just ask them if they have already written a review on the Chrome Webstore. 

See what I wrote above seems better. Too many questions are pointless.

> c. We can scrap this whole idea. I really do not mind. All new ideas present new issues. But I really do NOT mind scrapping.

I hope you meant scrapping to check if the user has written a review.

**Three** Ok, fine.

**Four** yes, that's good +1. I also expanded the option texts so that when the user clicks "Yes" he feels that "Yes I have checked" so that's a stronger statement kind of. When we speak something before clicking we feel more power attached to it. I think I am able to convey what I am trying to say?!?

**Comment by GaurangTandon** [comment]
So, what exactly was the summary? Do a follow-up comment here @iWrote

**Comment by iWrote** [comment]
Little feedback button appears after 20 days. 
Little beggar notice appears after 10 more days. 

User clicks button to see a big blue box. 

Two dongers appear. Their names are nice and meh in all caps.
NICE leads to review page. After taking permission to send user to a pub (lic review page.)
MEH leads to bug tantrum page.

Either way, the box disappears forever. Never to seek review again.

Solved Problems
- Use strong Calibri font.
- NICE must have a "already written a review" option.
- Not Now was reduced to simple red cross on upper-right corner of big blue box.

Unsolved Problems
- Reinstalls mean user will be prompted to write a review again, and again, and again. (And same goes for updates?)

**Comment by GaurangTandon** [comment]
> Unsolved Problems 1
Uhhmm, yes. Reinstalls => more prompts. Though I can set it up so that this set of data remains stored across reinstalls. Maybe after #93 , where similar feature is involved.
